libs/blocks/global-navigation/utilities/keyboard/mobilePopup.js

```diff
@@ -80,116 +79,157 @@ class Popup {
-          } else {
-            this.mobileArrowDown({ next });
-          }
-          break;
-        }
-        case 'Escape': {
-          closeAllDropdowns();
-          this.mainNav.items[this.mainNav.curr].focus();
-          break;
+  focusMainNav = (isFooter) => {
+    if (isFooter) return;
+    this.mainNav.items[this.mainNav.curr].focus();
+  };
+
+  focusMainNavNext = (isFooter) => {
+    if (isFooter) return;
+    this.mainNav.focusNext();
+    this.mainNav.open({});
+  };
+
+  handleKeyDown = ({ e, element, isFooter }) => {
+    const popupItems = [...element.querySelectorAll(selectors.popupItems)];
+    const curr = popupItems.findIndex((el) => el === e.target);
+    const prev = getPreviousVisibleItemPosition(curr, popupItems);
+    const next = getNextVisibleItemPosition(curr, popupItems);
+
+    e.preventDefault();
+    e.stopPropagation();
+
+    switch (e.code) {
+      case 'Tab': {
+        if (e.shiftKey) {
+          this.mobileArrowUp({ prev, curr, element, isFooter });
+        } else {
+          this.mobileArrowDown({ curr, next, element, isFooter });
         }
-        case 'ArrowLeft': {
-          const { prevHeadline, nextHeadline } = getState();
-          const headline = document.dir !== 'rtl' ? prevHeadline : nextHeadline;
-          if (!headline) {
-            closeHeadlines();
-            if (document.dir !== 'rtl') {
-              this.mainNav.items[this.mainNav.curr].focus();
-            } else {
-              this.mainNav.focusNext();
-              this.mainNav.open();
-            }
-            break;
+        break;
+      }
+      case 'Escape': {
+        closeAllDropdowns();
+        this.focusMainNav(isFooter);
+        break;
+      }
+      case 'ArrowLeft': {
+        const { prevHeadline, nextHeadline } = getState(element);
+        const headline = document.dir !== 'rtl' ? prevHeadline : nextHeadline;
+        if (!headline) {
+          if (!isFooter) closeHeadlines();
+          if (document.dir !== 'rtl') {
+            this.focusMainNav(isFooter);
+          } else {
+            this.focusMainNavNext(isFooter);
           }
-          openHeadline({ headline, focus: 'first' });
-          break;
-        }
-        case 'ArrowUp': {
-          this.mobileArrowUp({ prev, curr });
           break;
         }
-        case 'ArrowRight': {
-          const { prevHeadline, nextHeadline } = getState();
-          const headline = document.dir !== 'rtl' ? nextHeadline : prevHeadline;
-          if (!headline) {
-            closeHeadlines();
-            if (document.dir !== 'rtl') {
-              this.mainNav.focusNext();
-              this.mainNav.open();
-            } else {
-              this.mainNav.items[this.mainNav.curr].focus();
-            }
-            break;
+        openHeadline({ headline, focus: 'first' });
+        break;
+      }
+      case 'ArrowUp': {
+        this.mobileArrowUp({ prev, curr, element, isFooter });
```

Comment:
I think we need to add some CSS as well to highlight the active element, similar to the header.

header:
<img width="601" alt="Screenshot 2023-06-06 at 11 30 01" src="https://github.com/adobecom/milo/assets/408175/43b5b1c7-47a4-4448-971e-ca1aeefb9594">

footer:
<img width="597" alt="Screenshot 2023-06-06 at 11 30 16" src="https://github.com/adobecom/milo/assets/408175/f7ebaac3-85b1-4d9a-b052-b01ea996e675">

Reply:
That should work now!